### ISSUE TITLE
Indention failures after operator linebreaks

### ISSUE THREAD

**Issue by sempasha** [eslint/eslint#11776]
My environment:

* **ESLint Version:**
* **Node Version: 10.15.3**
* **npm Version: 6.9.0**

Parser version: default

<details>
<summary>Configuration</summary>

```js
{
  "env": {
    "es6": true,
    "mocha": true,
    "mongo": true,
    "node": true
  },
  "globals": {},
  "rules": {
    "indent": [
      2,
      4,
      {
        "SwitchCase": 1,
        "CallExpression": {
          "arguments": 1
        }
      }
    ],
    "operator-linebreak": [
      2,
      "after",
      {
        "overrides":
        {
          "?": "before",
          ":": "before"
        }
      }
    ]
  }
}
```

</details>

```js
test('Error occurs only when use new line after operator. ' +
    'Here is new line.',
    log);
```

```bash
eslint /path/to/test.js
```
I expect eslint check will succeed.
Actually it fails with error:

```
/path/to/test.js
  3:1  error  Expected indentation of 0 spaces but found 4  indent

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**Comment by kaicataldo** [comment]
Thanks for reporting this issue. Do you mind changing the example to the minimum repro case? It's hard to tell exactly which line the error is coming from right now.

**Comment by not-an-aardvark** [comment]
I think this is the same issue as https://github.com/eslint/eslint/issues/9706.

**Comment by sempasha** [comment]
@kaicataldo, done

**Comment by kaicataldo** [comment]
Thank you, that's very helpful. As @not-an-aardvark commented, it does look like the same aforementioned issue (see https://github.com/eslint/eslint/issues/9706#issuecomment-351290798 for the explanation as to why it works this way). 

What makes this tricky is that there are two things at play here: the indentation of the multiline expression as well as that of the arguments. While I agree that at first glance it doesn't look quite right, I do think it makes the most sense given the constraints. Do you have an alternative proposal for how you would expect the indent rule to enforce this?

I personally would expect either of the following to be enforced (based on if there were line breaks inside the parentheses):

```js
test('Error occurs only when use new line after operator. ' +
    'Here is new line.',
log);
```

```js
test(
    'Error occurs only when use new line after operator. ' +
        'Here is new line.',
    log
);
```

**Comment by sempasha** [comment]
**First. Single indention after operator `+`**

Expression - is a string declaration, expression consists of several strings with `+` operator as glue. Every line of this expression indented with single tab (4 spaces).
```js
let some = 'What makes this tricky is that there are two things at play here: ' +
    'the indentation of the multiline expression as well as that of the arguments. ' +
    'While I agree that at first glance it doesn't look quite right, ' +
    'I do think it makes the most sense given the constraints. ' +
    'Do you have an alternative proposal for ' +
    'how you would expect the indent rule to enforce this?';
```
I expect the same logic even it would be function call.
```js
doSome('What makes this tricky is that there are two things at play here: ' +
    'the indentation of the multiline expression as well as that of the arguments. ' +
    'While I agree that at first glance it doesn't look quite right, ' +
    'I do think it makes the most sense given the constraints. ' +
    'Do you have an alternative proposal for ' +
    'how you would expect the indent rule to enforce this?');
```
**Second. Single indention after comma in function call**
Expression is a function call with multiple arguments, every new line of expression indented with single tab (4 spaces).
```js
doSome(first,
  second,
  third);
```
**Third. Combination**
Expression is a function call with multiple arguments, every new line of expression indented with single tab (4 spaces). When one of parameters is a multi-line expression - every new line of this expression should be indented with extra tab (4 spaces), so indention will be double tab (8 spaces);

```js
doSome('What makes this tricky is that there are two things at play here: ' +
        'the indentation of the multiline expression as well as that of the arguments. ' +
        'While I agree that at first glance it doesn't look quite right, ' +
        'I do think it makes the most sense given the constraints. ' +
        'Do you have an alternative proposal for ' +
        'how you would expect the indent rule to enforce this?',
     second,
     third);

doSome(first,
    'What makes this tricky is that there are two things at play here: ' +
        'the indentation of the multiline expression as well as that of the arguments. ' +
        'While I agree that at first glance it doesn't look quite right, ' +
        'I do think it makes the most sense given the constraints. ' +
        'Do you have an alternative proposal for ' +
        'how you would expect the indent rule to enforce this?',
     third);
```
